### PR TITLE
restore iOS 12 support

### PR DESCRIPTION
**Pull request by aj-dt**
restore runtime support for iOS 12, tvOS 12, watchOS 4 and macOS 10.13.

Adds OS-sensitive versions for copying keys (`copyKey()`) and evaluating server trust (`evaluateCertificateChainTrust()`).

`copyKey()` calls `SecTrustCopyKey()` on iOS 14+ and `SecTrustCopyPublicKey()` on earlier OSs.

`evaluateCertificateChainTrust()` calls `SecTrustEvaluateWithError()` on iOS 12+ and `SecTrustEvaluate()` on earlier OSs.

Note that macOS, tvOS, watchOS have slightly different version thresholds.

One subtlety in the implementation lies in `evaluateCertificateChainTrust()` which effectively "returns" three values: the bool return value and the referenced parameters TrustResult and the NSError. Since `SecTrustEvaluateWithError()` and `SecTrustEvaluate()` return slightly different sets of these three values for the various possible scenarios, interpretation of these results is delicate and requires if conditions evaluating combinations of these values. I consider this complexity acceptable given that consumers of TrustKit are shielded from it. For validation of the current code we rely on the unit tests (which all pass). 

Note that with these changes support for **running** on iOS 12 and iOS 13 is restored, however **building** with old versions of Xcode may fail. In particular, the use of `@available(iOS 14.0)` likely introduces a requirement to build with Xcode >= 12 (the earliest Xcode to ship with iOS 14, see [here](https://developer.apple.com/support/xcode/))

### CODE REVIEW

**Comment by JuhaO81** [comment]
Thx for adding the support back! Much needed here :)

**Comment by aj-dt** [comment]
@nabla-c0d3 removed the bool return value from `evaluateCertificateChainTrust()` following your suggestion. It turned out to be redundant with the error parameter, so now there is less opportunity to cofuse that return value as a signal that the cert is trusted and one must examine the trust result to know the details.

File: TrustKit/Pinning/pinning_utils.m [hunk 1]
```diff
@@ -13,17 +13,34 @@
...
-        chainTrusted = false;
-        if (error != NULL) {
+void evaluateCertificateChainTrust(SecTrustRef serverTrust, SecTrustResultType *trustResult, NSError **error) {
+    if (@available(iOS 12.0, macOS 14.0, tvOS 12.0, watchOS 5.0, *)) {
+        CFErrorRef errorRef;
+        bool certificateEvaluationSucceeded = SecTrustEvaluateWithError(serverTrust, &errorRef);
+        OSStatus status = SecTrustGetTrustResult(serverTrust, trustResult);
+        if (status != errSecSuccess)
+        {
+            certificateEvaluationSucceeded = false;
```
Comment (nabla-c0d3) [inline TrustKit/Pinning/pinning_utils.m:23]: I think this line can be removed?

**Review by nabla-c0d3 — CHANGES_REQUESTED** [review]
One minor change but looks good overall. Thanks!

File: TrustKit/Pinning/pinning_utils.m (hunk 1 above)
Comment (aj-dt) [reply TrustKit/Pinning/pinning_utils.m:23]: Done

**Review by nabla-c0d3 — APPROVED** [review]

**Pull request merged**